### PR TITLE
Standalone Media Sample

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Standalone Media Sample

#### WHY

It's way bigger than the rest of the samples. And we want to make it even bigger.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: media-sample/src/main/java/com/google/android/horologist/mediasample/components/MediaActivity.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    override fun onResume() {
+        super.onResume()
+
+        lifecycleScope.launchWhenResumed {
+            val playerRepository = viewModelModule.playerRepository
+            val uampService =
+                viewModelModule.mediaApplicationModule.networkModule.uampService
+
+            playerRepository.connected.filter { it }.first()
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/components/MediaActivity.kt:70]: question: is this line necessary?

File: media-sample/src/main/java/com/google/android/horologist/mediasample/di/MediaActivityContainer.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import android.app.Activity
+import com.google.android.horologist.mediasample.components.MediaActivity
+import com.google.android.horologist.mediasample.components.MediaApplication
+
+/**
+ * Simple DI implementation - to be replaced by hilt.
+ */
+class MediaActivityContainer(
+    internal val mediaApplicationModule: MediaApplicationModule,
+    private val activity: MediaActivity,
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/di/MediaActivityContainer.kt:28]: nit: not in use

File: media-sample/src/main/java/com/google/android/horologist/mediasample/di/MediaApplicationModule.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.android.horologist.networks.data.DataRequestRepository
+import com.google.android.horologist.networks.status.NetworkRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+
+/**
+ * Simple DI implementation - to be replaced by hilt.
+ */
+class MediaApplicationModule(internal val application: MediaApplication) {
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/di/MediaApplicationModule.kt:42]: nit: sorry if my message was confused in the other PR but I think this class was fine being called MediaApplicationContainer given it has the application as scope? same for PlaybackServiceContainer

File: media-sample/src/main/java/com/google/android/horologist/mediasample/components/MediaActivity.kt (hunk 1 above)
Comment (yschimke) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/components/MediaActivity.kt:70]: Without this if you call setMediaItems too early it's a no-op.

```
    override fun setMediaItems(mediaItems: List<MediaItem>) {
        checkNotClosed()

        player.value?.let {
```

**Pull request merged**